### PR TITLE
Add article about caddy

### PR DESCRIPTION
**Pull request by YakoYakoYokuYoku**
## Description

This is a simple starting up guide for caddy development in Solus.

### Submitter Checklist

- [x] Updated the "lastmod" portion at the top of any modified Markdown files.
- [x] Squashed commits with `git rebase -i` (if needed)

### CODE REVIEW

File: software/caddy/en.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
++++
+# caddy
+
+Solus provides the caddy web server and proxy to users and developers who want to test their web applications before setting them to production.
+
+Remember that Solus is not a server operating system: any usage outside local testing is discouraged.
+
+## Installation
+
+caddy is available in the `caddy` package. It can be installed from the [Software Center](https://getsol.us/articles/common-tasks/installing-and-updating-software/en/) or via terminal:
```
Comment (JoshStrobl) [inline software/caddy/en.md:13]: URL should be relative e.g. `[Software Center](/articles/common-tasks/installing-and-updating-software/en/)`

File: software/caddy/en.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+```bash
```
Comment (JoshStrobl) [inline software/caddy/en.md:15]: whitespace before `bash` plz

File: software/caddy/en.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+sudo eopkg install caddy
+```
+
+## Usage
+
+### Configuration
+
+caddy, as many other packages in the Solus repository, is [stateless](https://clearlinux.org/features/stateless). That means the package gets installed with a vendored configuration out-of-the-box which can be customized, or even overridden, by a privileged user.
```
Comment (JoshStrobl) [inline software/caddy/en.md:23]: *like many other packages

File: software/caddy/en.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+caddy can use any valid configuration by reloading it via the cli. For automatic HTTPS users may need to generate their own certs and to add them at their `Caddyfile`s. For PHP FastCGI support `php` needs to be also installed and with the `php-fpm` service also running. Reloading with the `/usr/share/caddy/Caddyfile` reverts caddy to its original settings. More documentation about caddy is available at their [website](https://caddyserver.com/docs/).
+
+### Management
+
+Basic caddy management is done via `systemd` and the cli, using the following commands:
+
+```bash
```
Comment (JoshStrobl) [inline software/caddy/en.md:31]: whitespace before `bash` plz

**Review by JoshStrobl — CHANGES_REQUESTED** [review]

File: software/caddy/en.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+### Management
+
+Basic caddy management is done via `systemd` and the cli, using the following commands:
+
+    ```bash
+    sudo systemctl enable --now caddy # To enable nginx on startup and run it immediately.
+    caddy reload --config /etc/caddy/Caddyfile # To reload caddy with an specified configuration.
+    sudo systemctl stop caddy   # To stop caddy.
+    ```
```
Comment (JoshStrobl) [inline software/caddy/en.md:35]: Shouldn't be indenting and you never fixed the missing whitespace after `bash`

File: software/caddy/en.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+Remember that Solus is not a server operating system: any usage outside local testing is discouraged.
+
+## Installation
+
+caddy is available in the `caddy` package. It can be installed from the [Software Center](/articles/common-tasks/installing-and-updating-software/en/) or via terminal:
+
+    ```bash
+    sudo eopkg install caddy
+    ```
```
Comment (JoshStrobl) [inline software/caddy/en.md:17]: Shouldn't be indenting and you never fixed the missing whitespace after `bash`

**Review by JoshStrobl — CHANGES_REQUESTED** [review]

**Comment by JoshStrobl** [comment]
I believe there is a misunderstanding. I'm not talking about adding whitespace before all the scripting, but before the literal word `bash`.

Example:

``` markdown
``` bash
```

File: software/caddy/en.md (hunk 6 above)
Comment (YakoYakoYokuYoku) [reply software/caddy/en.md]: Yeah, seems I've forgot about the Markdown syntax for selecting the language

**Comment by JoshStrobl** [comment]
LGTM, thanks!

**Pull request merged**